pkg/util/duration/duration.go

```diff
@@ -617,16 +617,34 @@ func (d Duration) MulFloat(x float64) Duration {
 
 // DivFloat returns a Duration representing a time length of d/x.
 func (d Duration) DivFloat(x float64) Duration {
-	monthInt, monthFrac := math.Modf(float64(d.Months) / x)
-	dayInt, dayFrac := math.Modf((float64(d.Days) / x) + (monthFrac * DaysPerMonth))
+	// In order to keep it compatible with PostgreSQL, we use the same logic.
+	// Refer to https://github.com/postgres/postgres/blob/e56bce5d43789cce95d099554ae9593ada92b3b7/src/backend/utils/adt/timestamp.c#L3266-L3304.
+	month := int32(float64(d.Months) / x)
+	day := int32(float64(d.Days) / x)
+
+	monthRemainerDays := (float64(d.Months)/x - float64(month)) * DaysPerMonth
+	monthRemainerDays = tsround(monthRemainerDays)
+	secRemainder := (float64(d.Days)/x - float64(day) +
+		monthRemainerDays - float64(int64(monthRemainerDays))) * SecsPerDay
+	secRemainder = tsround(secRemainder)
+	if math.Abs(secRemainder) >= SecsPerDay {
+		day += int32(secRemainder / SecsPerDay)
+		secRemainder -= float64(int32(secRemainder/SecsPerDay) * SecsPerDay)
+	}
+	day += int32(monthRemainerDays)
+	ret := time.Duration(int64(math.RoundToEven(float64(time.Duration(d.nanos).Microseconds())/x+secRemainder*MicrosPerMilli*MillisPerSec))) * time.Microsecond
```

Comment:
this is pretty hard to read, can we break it down?

```
secs := time.Duration(math.RoundToEven(float64(time.Duration(d.nanos).Microseconds()) / x) * time.Microsecond
secs += time.Duration(secRemainder * time.Second)
```

Reply:
I think it should be break it down to``